AI/VCAI/VCAI.cpp

```diff
@@ -1007,6 +1008,9 @@ void VCAI::mainLoop()
 				for (auto basicGoal : ultimateGoalsFromBasic[goalToRealize])
 					goalsToRemove.push_back(basicGoal);
 
+				// sometimes resource manager contains an elementar goal which is not able to execute anymore and just fails each turn.
+				ai->ah->notifyGoalCompleted(goalToRealize);

```

Comment:
Just an optimization hack. Otherwise it will try to recruit hero once each turn until is completed.